### PR TITLE
fix(storage): accept google storage bucket urls

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

In line with Firebase Storage http logic on `firebase-js-sdk`.

## Related Issues

fixes https://github.com/FirebaseExtended/flutterfire/issues/6773

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_storage/firebase_storage/lib/src/utils.dart [hunk 1]
```diff
@@ -31,22 +31,48 @@ Map<String, String?>? partsFromHttpUrl(String url) {
...
+  String bucketDomain = r'([A-Za-z0-9.\-_]+)';
 
-  Uri uri = Uri.parse(decodedUrl);
+  if (decodedUrl.contains(firebaseStorageHost)) {
+    String version = 'v[A-Za-z0-9_]+';
+    String firebaseStoragePath = r'(/([^?#]*).*)?$';
+    RegExp firebaseStorageRegExp = RegExp(
+      '^https?://$firebaseStorageHost/$version/b/$bucketDomain/o$firebaseStoragePath',
+      caseSensitive: false,
+    );
```
Comment (lesnitsky) [inline packages/firebase_storage/firebase_storage/lib/src/utils.dart:46]: any chance to make these strings and regexp static/const?

Comment (russellwheatley) [reply packages/firebase_storage/firebase_storage/lib/src/utils.dart:46]: I've made strings `const`, cannot make `RegExp` const.

Comment (lesnitsky) [reply packages/firebase_storage/firebase_storage/lib/src/utils.dart:46]: These strings and regexp could be moved outside of the function body, so they'll be created only once, not on each function invocation

**Review by lesnitsky — APPROVED** [review]
lgtm

**Pull request merged**